core/blockly.js

```diff
@@ -405,9 +405,15 @@ Blockly.jsonInitFactory_ = function(jsonDef) {
  */
 Blockly.defineBlocksWithJsonArray = function(jsonArray) {
   for (var i = 0, elem; elem = jsonArray[i]; i++) {
-    Blockly.Blocks[elem.type] = {
-      init: Blockly.jsonInitFactory_(elem)
-    };
+    var typename = elem.type;
+    if (typename == null || typename === "") {
```

Comment:
Single quotes preferred.  But this should just be:
  if (elem.type) {
In which case no need for a variable since it would only be used here once, and once below.